api/segments/migrations/0028_condition_property_required.py

```diff
@@ -0,0 +1,69 @@
+from collections.abc import Callable
+from typing import Any
+
+from django.apps.registry import Apps
+from django.db import migrations, models
+
+
+def fill_percentage_split_property_for(model_name: str) -> Callable[[Apps, Any], None]:
+    def fill_property(apps: Apps, _: Any) -> None:
+        model = apps.get_model("segments", model_name)
+        model.objects.filter(
+            models.Q(operator="PERCENTAGE_SPLIT") & (
+                models.Q(property__isnull=True) | models.Q(property__regex=r"^\s*$")
```

Comment:
```suggestion
                models.Q(property__isnull=True) | models.Q(property="")
```

Is there something I'm missing?

Reply:
Any evidence of such values (e.g. from Metabase)?

Reply:
No evidence from the SaaS standpoint, though I preferred to stay on the safer side and also cover `null`s and other empty-ish that could occur in private installs for whatever reason. TBH not sure this is worth the database CPU effort in the migration.

Reply:
> No evidence from the SaaS standpoint

Good to know, thanks! I think it's a good argument for applying the suggestion as well.

> not sure this is worth the database CPU effort in the migration but definitely safer.

I'm strictly on the side of performant rather than redundant data migrations.